Tests fail with new Docker version scheme

@thomasleveil: Docker has reworked their versioning system and I have installed the latest version in Ubuntu 16.04, which is versioned `Docker version 17.03.0-ce, build 60ccb22` (from the official repo).

This causes a failure in the `pytest` test suite (stripped wide header lines):
```
root@f8107fadcf50:/home/steve/Workspace/nginx-proxy/test# pytest
platform linux2 -- Python 2.7.13, pytest-3.0.5, py-1.4.32, pluggy-0.4.0 -- /usr/local/bin/python2
rootdir: /home/steve/Workspace/nginx-proxy/test, inifile: pytest.ini
collected 101 items / 1 errors 

test_dockergen/test_dockergen_v3.py:12: in <module>
    pytestmark = pytest.mark.skipif(versiontuple(docker_version) < versiontuple('1.13'),
test_dockergen/test_dockergen_v3.py:8: in versiontuple
    return tuple(map(int, (v.split("."))))
E   ValueError: invalid literal for int() with base 10: '0-ce'
```

The problem is that the version now contains `-ce`.  My understanding is that version `17.03` (they use date-based versioning now) is equivalent to either `1.13` or the non-existent `1.14`.  In any case, this is the code that is causing the failure: https://github.com/jwilder/nginx-proxy/blob/master/test/test_dockergen/test_dockergen_v3.py#L12-L13

I am reworking my SSL Enhancement tests at the moment, so I will work around this for the time being, but if you (@thomasleveil) could look into it, I would be grateful.  Otherwise, I can probably fix it when I have some more time (this weekend, perhaps).

Until this issue is fixed, the unit test will not work on Docker `> 1.13`.

--- a/test/test_dockergen/test_dockergen_v3.py
+++ b/test/test_dockergen/test_dockergen_v3.py
@@ -5,12 +5,26 @@ import pytest
 
 
 def versiontuple(v):
-    return tuple(map(int, (v.split("."))))
+    """
+    >>> versiontuple("1.12.3")
+    (1, 12, 3)
+
+    >>> versiontuple("1.13.0")
+    (1, 13, 0)
+
+    >>> versiontuple("17.03.0-ce")
+    (17, 3, 0)
+
+    >>> versiontuple("17.03.0-ce") < (1, 13)
+    False
+    """
+    return tuple(map(int, (v.split('-')[0].split("."))))
 
 
-docker_version = docker.from_env().version()['Version']
-pytestmark = pytest.mark.skipif(versiontuple(docker_version) < versiontuple('1.13'),
-                                reason="Docker compose syntax v3 requires docker engine v1.13")
+raw_version = docker.from_env().version()['Version']
+pytestmark = pytest.mark.skipif(
+    versiontuple(raw_version) < (1, 13),
+    reason="Docker compose syntax v3 requires docker engine v1.13 or later (got %s)" % raw_version)
 
 
 @pytest.yield_fixture(scope="module")
@@ -45,3 +59,8 @@ def test_forwards_to_whoami(nginx_tmpl, docker_compose, nginxproxy):
     assert r.status_code == 200
     whoami_container = docker_compose.containers.get("whoami")
     assert r.text == "I'm %s\n" % whoami_container.id[:12]
+
+
+if __name__ == '__main__':
+    import doctest
+    doctest.testmod()

